### PR TITLE
Added command line argument for IntervalMerging based on GATK3

### PR DESCRIPTION
**Pull request by jamesemery**
As well as some tests to the test suite.

### CODE REVIEW

**Comment by jamesemery** [comment]
@droazen

File: src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java [hunk 1]
```diff
@@ -84,6 +83,13 @@
      * should be returned. Lazily initialized.
      */
     protected TraversalParameters traversalParameters = null;
+    /**
+     * By default, the program merges abutting intervals (i.e. intervals that are directly side-by-side but do not
+     * actually overlap) into a single continuous interval. However you can change this behavior if you want them to be
+     * treated as separate intervals instead.
+     */
+    @Argument(fullName = "interval_merging", shortName = "im", doc = "Interval merging rule for abutting intervals", optional = true) //TODO
+    protected IntervalMergingRule intervalMerging = IntervalMergingRule.ALL;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java]: Move this above the start of the non-argument fields for the class

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java]: Also remove the TODO

File: src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java [hunk 2]
```diff
@@ -84,6 +83,13 @@
      * should be returned. Lazily initialized.
      */
     protected TraversalParameters traversalParameters = null;
+    /**
+     * By default, the program merges abutting intervals (i.e. intervals that are directly side-by-side but do not
+     * actually overlap) into a single continuous interval. However you can change this behavior if you want them to be
+     * treated as separate intervals instead.
+     */
+    @Argument(fullName = "interval_merging", shortName = "im", doc = "Interval merging rule for abutting intervals", optional = true) //TODO
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollection.java]: Rename to `interval_merging_rule` and `imr`

File: src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java [hunk 3]
```diff
@@ -92,6 +93,22 @@ public void testIncludeWithPadding(IntervalArgumentCollection iac){
...
+        iac.intervalMerging = IntervalMergingRule.ALL;
+        Assert.assertEquals(iac.getIntervals(hg19GenomeLocParser.getSequenceDictionary()), Arrays.asList(new SimpleInterval("1", 1, 200)));
+    }
+
+    @Test(dataProvider = "optionalOrNot")
+    public void testIntervalMergingRuleAdjacentNoMerge(IntervalArgumentCollection iac){
+        iac.addToIntervalStrings("1:1-100");
+        iac.addToIntervalStrings("1:101-200");
+        iac.intervalMerging = IntervalMergingRule.OVERLAPPING_ONLY;
+        Assert.assertEquals(iac.getIntervals(hg19GenomeLocParser.getSequenceDictionary()).size(), 2);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java]: I'd also assert that the output intervals are the same as the input intervals

Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java]: Also, indentation issue here.

File: src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java [hunk 4]
```diff
@@ -92,6 +93,22 @@ public void testIncludeWithPadding(IntervalArgumentCollection iac){
...
+    }
+
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java]: Add a test case that uses the INTERSECTION set rule in combination with both of the merging rules

Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java]: (ie., prove that the set rule is applied before the merge rule)

File: src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java (hunk 3 above)
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/cmdline/argumentcollections/IntervalArgumentCollectionTest.java]: Responded to the comments

**Pull request merged**